dev/src/util.ts

```diff
@@ -43,3 +48,49 @@ export function autoId(): string {
 export function requestTag(): string {
   return autoId().substr(0, 5);
 }
+
+/**
+ * Parses the arguments for the `getAll()` call supported by both the Firestore
+ * and Transaction class.
+ *
+ * @private
+ * @param validator The argument validator to use.
+ * @param documentRefsOrReadOptions An array of document references followed by
+ * an optional ReadOptions object.
+ */
+export function parseGetAllArguments(
+    validator: AnyDuringMigration,
+    documentRefsOrReadOptions: Array<DocumentReference|ReadOptions>):
+    {documents: DocumentReference[], fieldMask: FieldPath[]|null} {
+  let documents: DocumentReference[];
+  let readOptions: ReadOptions|undefined = undefined;
+
+  const usesVarags = !Array.isArray(documentRefsOrReadOptions[0]);
+
+  if (usesVarags) {
+    if (documentRefsOrReadOptions.length > 0 &&
+        isPlainObject(
+            documentRefsOrReadOptions[documentRefsOrReadOptions.length - 1])) {
+      readOptions = documentRefsOrReadOptions.pop() as ReadOptions;
+      documents = documentRefsOrReadOptions as DocumentReference[];
+    } else {
+      documents = documentRefsOrReadOptions as DocumentReference[];
+    }
+  } else {
+    // Support an array of document references as the first argument for
```

Comment:
Is there a test for this legacy support? I would not expect a call that hits this condition to typecheck.

Reply:
Found the test, and it looks like it wouldn't typecheck without a cast.

Reply:
Yeah, it's meant to support users that still use JS directly *shudder* :)